tests/Tests/500__jupyterhub/culler.robot

```diff
@@ -0,0 +1,169 @@
+    [Documentation]    Spawn a minimal image
+    Begin Web Test
+    Launch JupyterHub Spawner From Dashboard
+    Spawn Notebook With Arguments  image=s2i-minimal-notebook  size=Default
+
+Get Culler Pod
+    [Documentation]    Finds the current culler pod and returns the name
+    ${culler_pod} =  OpenShiftLibrary.Oc Get  kind=Pod  label_selector=app=jupyterhub-idle-culler  namespace=redhat-ods-applications
+    ${length} =  Get Length  ${culler_pod}
+    # Only 1 culler pod, correct one
+    IF  ${length}==1
+        ${culler_pod_name} =  Set Variable  ${culler_pod[0]}[metadata][name]
+    ELSE
+    # There can be more than one during rollout
+        Sleep  10s
+        ${culler_pod_name} =  Get Culler Pod
+    END
+    Log  ${culler_pod}
+    Log  ${culler_pod_name}
+    [Return]  ${culler_pod_name}
+
+Get Culler Timeout
+    [Documentation]    Gets the current culler timeout
+    ${current_timeout} =  OpenShiftLibrary.Oc Get  kind=ConfigMap  name=jupyterhub-cfg  namespace=redhat-ods-applications  fields=['data.culler_timeout']
+    ${current_timeout} =  Set Variable  ${current_timeout[0]['data.culler_timeout']}
+    Log  ${current_timeout}
+    Log To Console  ${current_timeout}
+    ${CULLER_POD} =  Get Culler Pod
+    ${culler-env-timeout} =  Run  oc exec ${CULLER_POD} -n redhat-ods-applications -- printenv CULLER_TIMEOUT
+    Should Be Equal  ${current_timeout}  ${culler-env-timeout}
+    [Return]  ${current_timeout}
+
+Modify Culler Timeout
+    [Documentation]    Modifies the culler timeout via UI
+    [Arguments]    ${new_timeout}
+    Open Dashboard Cluster Settings
+    Set Timeout To  ${new_timeout}
+    # Click button to save/apply the new timeout (need final UI)
+    # Enough time to start the rollout
+    Sleep  60s
+
+Open Dashboard Cluster Settings
+    [Documentation]    Opens the RHODS dashboard and navigates to the Cluster settings page
+    Set Library Search Order    SeleniumLibrary
+    Launch Dashboard    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+    ...    ${ODH_DASHBOARD_URL}    ${BROWSER.NAME}    ${BROWSER.OPTIONS}
+    Sleep  5
+    ${settings_hidden} =  Run Keyword And Return Status  Page Should Contain Element  xpath://section[@aria-labelledby="settings"][@hidden=""]
+    IF  ${settings_hidden}==True
+        Click Element  xpath://button[@id="settings"]
+    END
+    Click Element  xpath://a[.="Cluster settings"]
+
+Set Timeout To
+    [Documentation]    Helper to modify culler timeout via UI
+    [Arguments]    ${new_timeout}
+    ${hours}  ${minutes} =  Convert To Hours And Minutes  ${new_timeout}
+    Sleep  5
+    ${disabled_field} =  Run Keyword And Return Status  Page Should Contain Element  //input[@id="hour-input"][@disabled=""]
+    IF  ${disabled_field}==True
+        Click Element  xpath://input[@id="culler-timeout-limited"]
+    END
+    Input Text  //input[@id="hour-input"]  ${hours}
+    Input Text  //input[@id="minute-input"]  ${minutes}
+
+Disable Culler
```

Comment:
This keyword could be useful in ODHDashboard.robot. It could be renamed to Disable Notebook Culler

Reply:
I'm going to answer to one but it applies to all comments about moving these to different Resource folders/files: Do we expect other TCs to actually use anything related to the NB culler? Or will any additional culler TC always end up in this `culler.robot` file? Because if so IMHO it makes more sense to keep these keywords here, with maybe the exception of `Open Dashboard Cluster Settings`

Reply:
In #304  there is test ODS-547 that is testing notebook culling for Sandbox. It's waiting for 24h, but if it could set the timeout for a shorter time it could be better

Reply:
Isn't the sandbox just a different environment with the same rhods build? If that is the case, apart from sandbox-specific test cases, this file as is would already be able to test the culler functionality in the sandbox environment

Reply:
It's ok for me. @tarukumar do you agree?

Reply:
@tarukumar What do you think about this?

Reply:
Renamed the keyword